explainer-content-size.md

```diff
@@ -0,0 +1,34 @@
+
+This proposal exposes these intrinsic sizes explicitly. (In that sense, one motivation for the `content-size` property could be [explaining](https://extensiblewebmanifesto.org) these algorithms. One practical application could be its use in testing their behavior.
+
+## Application to non-rendered subtrees
+
+A more compelling and direct motivation for exposing `content-size` is to use it in cases when the layout of a subtree is *not available*. The layout can be unavailable in situations such as:
+* The subtree is not yet fully loaded from the network, or rendered into DOM by a custom element or framework
+* The User Agent has skipped rendering for the subtree as an optimization
+
+Note that these two cases can be seen as the same use case, if the concept of "rendering" is extended to include asynchronous, scheduled factors such as the network, a DOM rendering framework [1], a [custom element](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements), dynamic replaced elements such as embedded SVG documents, images and videos. In other words, the [rendering event loop](https://github.com/chrishtr/rendering/blob/master/rendering-event-loop.md) is only one source of potential asynchrony in an expanded concept of rendering.
+
+In such cases, the developer may still wish for the element to participate in the intrinsic sizing aspects of layout, by providing placeholder sizing, because otherwise there may be an undesirable, jarring [shift of the layout](https://web.dev/layout-instability-api) of the page as the content renders.
+
+In addition, the height of an overflow-scrolling region depends on the [layout overflow size](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Flow_Layout/Flow_Layout_and_Overflow) of its content, which includes intrinsic sizing. This allows a web developer to avoid spending time rendering scrollable content which is not visible to the user, but still have it take up space in the scroller.
+
+See also the [main explainer](https://github.com/WICG/display-locking/blob/master/README.md) for display-locking for more motivating examples along these lines.
+
+## Examples of `content-size`
+
+[Here](https://wicg.github.io/display-locking/sample-code/contain-size-block-flow-examples.html) is a page demonstrating some simple block-flow layout use-cases.
+
+[Here](https://wicg.github.io/display-locking/sample-code/contain-size-flexbox-examples.html) is a page demonstrating a basic flexbox use-case.
```

Comment:
The last example in this link should say "with min-width/min-height", and also the style of it has a typo where it say "minheight" instead of "min-height"

Reply:
Done.